mobify/progressive-web-scaffold
# Data Architecture 2.0: Checkout Confirmation

**JIRA**: [WEB-1113](https://mobify.atlassian.net/browse/WEB-1113)

## Changes
- Fetch data from confirmation page, or from redux store based on data from previous steps
- Add functionality to the confirmation page's "Registration" form
    - Registers the user
    - Updates the user's address book with information received in previous steps

## Todos
- [x] Create actions, et al. for submitting Registration form
    - [x] Hide registration form on success
    - [x] After successful registration, Log user in
- ~~Add "on mount" login detection (should persist from Shipping/Payment)~~
- [x] Retrieve order # from page
- [x] Retrieve order URL (if logged in)

## Post _"Integration with Shipping & Payment"_ Todo
- [x] Fix router path to `checkout/onepage/success/` to enable proper fetching of content
- [x] Verify that when logged out, arriving on the Confirmation Page also correctly resets the Cart count to zero

## How to test-drive this PR
- Run `npm run dev`
- Preview [Merlin's Potions](https://preview.mobify.com/?url=https%3A%2F%2Fwww.merlinspotions.com%2F&site_folder=https%3A%2F%2Flocalhost%3A8443%2Floader.js&disabled=0&domain=&scope=1)
- add an item to the cart
- proceed through the checkout to the confirmation page
- confirm the confirmation page works as expected for logged in users and guest users
